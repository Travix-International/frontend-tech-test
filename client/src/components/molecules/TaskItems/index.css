.task-items {
  display: flex;
  flex: 1;
  flex-direction: column;
  margin: 0;
  padding: 0; }
