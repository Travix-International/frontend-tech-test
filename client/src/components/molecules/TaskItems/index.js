import React from 'react'
import PropTypes from 'prop-types'
import './index.css'

import TaskItem from '../../molecules/TaskItem/index'

const TaskItems = ({ items, updateTask, deleteTask }) => {
  return (
    <ul className='task-items'>
      {
        items.map((task) => {
          return (
            <TaskItem
              key={task.id}
              id={task.id}
              title={task.title}
              description={task.description}
              updateTask={updateTask}
              deleteTask={() => { deleteTask({ id: task.id }) }}
            />
          )
        })
      }
    </ul>
  )
}

TaskItems.propTypes = {
  items: PropTypes.array.isRequired,
  updateTask: PropTypes.func.isRequired,
  deleteTask: PropTypes.func.isRequired
}

export default TaskItems
