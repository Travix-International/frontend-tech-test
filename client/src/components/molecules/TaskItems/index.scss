@import 'src/styles/base/colors.scss';
@import 'src/styles/base/medias.scss';

.task-items {
  background-color: $white;
  display: flex;
  flex: 1;
  flex-direction: column;
}

.task-items-list {
  display: flex;
  flex-direction: column-reverse;
  flex-grow: 1;
  height: 100%;
  margin: 0;
  padding: 0;
}
