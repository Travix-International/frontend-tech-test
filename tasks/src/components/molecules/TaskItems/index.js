import React from 'react'
import PropTypes from 'prop-types'

import TaskItem from '../../molecules/TaskItem/index'

const TaskItems = ({ items, updateTask, deleteTask }) => {
  return (
    <ul className='task-list'>
      {
        items.map((task) => {
          return (
            <TaskItem
              key={task.id}
              id={task.id}
              title={task.title}
              description={task.description}
              updateTask={updateTask}
              deleteTask={() => { deleteTask({ id: task.id }) }}
            />
          )
        })
      }
    </ul>
  )
}

TaskItems.propTypes = {
  items: PropTypes.array.isRequired,
  updateTask: PropTypes.func.isRequired,
  deleteTask: PropTypes.func.isRequired
}

export default TaskItems
