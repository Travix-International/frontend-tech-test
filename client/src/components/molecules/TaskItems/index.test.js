import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import TaskItems from './index'

describe('TaskItems', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer()
    const tasks = [
      { id: 1, title: 'A title', description: 'A description' }
    ]
    const updateTask = jest.fn()
    const deleteTask = jest.fn()

    renderer.render(
      <TaskItems
        items={tasks}
        updateTask={updateTask}
        deleteTask={deleteTask}
      />
    )
    const result = renderer.getRenderOutput()

    expect(result).toMatchSnapshot()
  })

  it('should call passed updateTask when child updateTask is called', () => {
    const renderer = new ShallowRenderer()
    const tasks = [
      { id: 1, title: 'A title', description: 'A description' }
    ]
    const updateTask = jest.fn()
    const deleteTask = jest.fn()

    renderer.render(
      <TaskItems
        items={tasks}
        updateTask={updateTask}
        deleteTask={deleteTask}
      />
    )
    const result = renderer.getRenderOutput()
    const taskItem = result.props.children[0]

    expect(taskItem.props.updateTask('payload'))
    expect(updateTask).toBeCalledWith('payload')
  })

  it('should call passed deleteTask when child deleteTask is called', () => {
    const renderer = new ShallowRenderer()
    const tasks = [
      { id: 1, title: 'A title', description: 'A description' }
    ]
    const updateTask = jest.fn()
    const deleteTask = jest.fn()

    renderer.render(
      <TaskItems
        items={tasks}
        updateTask={updateTask}
        deleteTask={deleteTask}
      />
    )
    const result = renderer.getRenderOutput()
    const taskItem = result.props.children[0]

    expect(taskItem.props.deleteTask())
    expect(deleteTask).toBeCalledWith({ id: 1 })
  })
})
