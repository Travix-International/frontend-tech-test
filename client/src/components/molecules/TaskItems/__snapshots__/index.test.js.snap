// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItems should render correctly with active loading 1`] = `<LoadingIndicator />`;

exports[`TaskItems should render correctly with inactive loading 1`] = `
<Unknown
  deleteTask={[Function]}
  items={
    Array [
      Object {
        "description": "A description",
        "id": 1,
        "title": "A title",
      },
    ]
  }
  loading={false}
  updateTask={[Function]}
/>
`;
